### PR TITLE
[onnx] Add support for torch.cross and torch.cdist

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Add support for following operators:
- torch.cross
- torch.linalg.cross
- torch.cdist
- torch.nn.pairwisedistance

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -3462,6 +3474,35 @@ def roll(g, self, shifts, dims):
...
+    # A' = [c, a, b], B' = [e, f, d], so that we calculate (c*e, a*f, b*d)
+    roll_x_2 = roll(g, input, [1], [dim])
+    roll_y_2 = roll(g, other, [2], [dim])
+    # cross product is calculated as
+    # result = [(b*f - c*e), (c*d - a*f), (a*e - b*d)]
+    return sub(g, mul(g, roll_x_1, roll_y_1), mul(g, roll_x_2, roll_y_2))
+
+
+def cdist(g, x1, x2, p=2.0, compute_mode="use_mm_for_euclid_dist_if_necessary"):
+    # X1.shape = (B * P * D), X2.shape = (B * R * D)
```
Comment (vadimkantorov) [inline torch/onnx/symbolic_opset9.py:3496]: if compute_mode is not respected, should there be some notes about this in docs?

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -1562,6 +1562,13 @@ def cosine_similarity(g, x1, x2, dim, eps):
     return div(g, cross, div_tens)
 
 
+def pairwise_distance(g, input1, input2, p, eps, keepdim):
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1565]: `eps` is unused?

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -3462,6 +3474,37 @@ def roll(g, self, shifts, dims):
...
+    # X1.shape = (B * P * D), X2.shape = (B * R * D)
+    # In order to respect numpy style broadcasting as demonstrated in
+    # https://github.com/onnx/onnx/blob/main/docs/Broadcasting.md
+    # we unsqueeze both input tensors
+    # Currently we ignore the 'compute_mode' variable as we use default to
+    # using matrix multiplication to calculate the euclidean distance
+    rank = sym_help._get_tensor_rank(x1)
+    broadcasted_x1 = sym_help._unsqueeze_helper(g, x1, [rank - 1])
+    broadcasted_x2 = sym_help._unsqueeze_helper(g, x2, [rank - 2])
+    return pairwise_distance(g, broadcasted_x1, broadcasted_x2, p, eps=1e-06, keepdim=False)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:3505]: would both broadcasted tensor have rank 4? how does it work with `pairwise_distance`?

**Review by BowenBao — COMMENTED** [review]
Please rebase with master to resolve CI.

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:1565]: Added

**Review by BowenBao — APPROVED** [review]

**Comment by BowenBao** [comment]
@shubhambhokare1 please check onnx check failures.

**Pull request closed without merging**